zds/tutorialv2/publication_utils.py

```diff
@@ -346,23 +348,22 @@ def pdftex(self, texfile):
         except ImportError:
             pass
 
-        if exit_code == 0 and os.exists(texfile[:-3] + "pdf"):
-            # success
-            return True
-        self.handle_pdf_tex_error(new_err.getvalue())
+        return "fatal" not in output_err.decode('utf-8') and os.path.exists(texfile[:-3] + "pdf")
 
     def make_glossary(self, basename, texfile):
-        new_err = six.StringIO()
-        exit_code = subprocess.call('makeglossaries ' + basename,
-                                    shell=True, cwd=os.path.dirname(texfile), stderr=new_err)
+        command_process = subprocess.Popen('makeglossaries ' + basename,
+                                           shell=True, cwd=os.path.dirname(texfile),
+                                           stdout=subprocess.PIPE,
+                                           stderr=subprocess.PIPE)
+        std_out, std_err = command_process.communicate()
         try:
             from raven import breadcrumbs
             breadcrumbs.record(message="makeglossaries call",
                                data='makeglossaries ' + basename,
                                type="cmd")
         except ImportError:
             pass
-        if exit_code == 0:
+        if "fatal" not in std_out.decode('utf-8').lower() and 'fatal' not in std_err.decode('utf-8').lower():
```

Comment:
Est-ce que tu peux stp ajouter un TODO ici aussi, à moins que tu aies étudié les exit code de makeglossaries et déterminé que le moyen le moins sale était `"fatal" not in` ?